torch/csrc/jit/script/compiler.cpp

```diff
@@ -1360,20 +1366,55 @@ struct to_ir {
     return num_starred;
   }
 
-  void emitAssignment(const Assign& stmt) {
-    bool starred_unpack = calcNumStarredUnpack(stmt.lhs(), stmt.range());
-    if (stmt.reduction() != '=') {
-      if (stmt.lhs().size() != 1) {
-        throw ErrorReport(stmt)
-            << "reductions are only allowed when there is a single variable "
-            << "on the left-hand side.";
+  // Emit nodes for augmented assignments like `+=`
+  void emitAugAssignment(const AugAssign& stmt) {
+    auto lhs = Var(stmt.lhs());
+    auto lhsValue = environment_stack->getSugaredVar(lhs.name())
```

Comment:
I would prefer this used `emitExpr` rather do lookup in the environment stack directly. If we ever change how TK_VAR is handled, then we might forget to edit this location.

Reply:
Ditto above